### PR TITLE
Correct documentation on operation 'complete'

### PR DESCRIPTION
**Pull request by Gormartsen**
Relate to #1432

@stephenplusplus My pardon! 
operation.on('complete') return metadata only. THere is no err value.

I decided to add link to code where all operations emits described.

Feel free to request updates to this PR. No offence here.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -393,14 +393,13 @@ var name = 'ubuntu-http';
 zone.createVM(name, { os: 'ubuntu' }, function(err, vm, operation) {
   // `operation` lets you check the status of long-running tasks.
 
-  operation.on('complete', function(err, metadata) {
-    if (!err) {
-      // Virtual machine created!
-    }
+  operation.on('complete', function(metadata) {
```
Comment (stephenplusplus) [inline README.md]: Woops, my bad for not catching that. You can show how to listen to an error:

```js
operation
  .on('error', function(err) {})
  .on('complete', function(metadata) {
    // Virtual machine created!
  });
```

Comment (Gormartsen) [inline README.md]: Will add it now

File: README.md [hunk 2]
```diff
@@ -393,14 +393,13 @@ var name = 'ubuntu-http';
...
+    // Virtual machine created!
   });
 });
 ```
 
+You can read more about opeation events [here](https://github.com/GoogleCloudPlatform/gcloud-node/blob/276c847669470c3639eb78d7d87a09db7a9b5c18/lib/compute/operation.js#L83).
```
Comment (stephenplusplus) [inline README.md]: You can link to the more readable version here: https://googlecloudplatform.github.io/gcloud-node/#/docs/compute/operation

Comment (stephenplusplus) [inline README.md]: Sorry, I think we should skip this link. It goes against the pattern of our other API doc sections. In general, if a user wants more info about an API, they can click the API or Official Documentation links above the code example.

Comment (Gormartsen) [inline README.md]: I will just put all options for operation. Is that better ?

Comment (stephenplusplus) [inline README.md]: What do you mean?

Comment (Gormartsen) [inline README.md]:
```
  operation
  .on('error', function(err) {})
  .on('running', function(metadata) {})
  .on('complete', function(metadata) {
    // Virtual machine created!
  });
```

Comment (stephenplusplus) [inline README.md]: Oh gotcha. Yeah, that's perfect. Thanks!

Comment (Gormartsen) [inline README.md]: Looks better now! Cheers! Going back to coding.

File: README.md [hunk 3]
```diff
@@ -393,15 +393,15 @@ var name = 'ubuntu-http';
 zone.createVM(name, { os: 'ubuntu' }, function(err, vm, operation) {
   // `operation` lets you check the status of long-running tasks.
 
-  operation.on('complete', function(err, metadata) {
-    if (!err) {
-      // Virtual machine created!
-    }
+  operation
+  .on('error', function(err) {})
```
Comment (stephenplusplus) [inline README.md]: Just two spaces on each line after `operation`, please:

```js
operation
  .on('error', function(err) {})
  .on('running', function(metadata) {})
  .on('complete', function(metadata) {
    // Virtual machine created!
  });
```

File: README.md (hunk 2 above)
Comment (stephenplusplus) [inline README.md]: Oh man, so sorry, I had one last small thing about indenting. Just let me know if you want me to get it. I appreciate the help!

File: README.md (hunk 3 above)
Comment (Gormartsen) [inline README.md]: my bad!

**Comment by stephenplusplus** [comment]
Thank you very much! Happy coding! 💻 ☕

**Pull request merged**